tests/unit/s2n_self_talk_client_hello_cb_test.c

```diff
@@ -489,8 +489,7 @@ int run_test_poll_ch_cb(s2n_client_hello_cb_mode cb_mode,
     struct s2n_test_io_pair io_pair = { 0 };
     struct s2n_config *config = s2n_config_new();
     EXPECT_NOT_NULL(config);
-    struct s2n_connection *conn = s2n_connection_new(S2N_SERVER);
-    EXPECT_NOT_NULL(conn);
+    struct s2n_connection *conn;
```

Comment:
Got memory leak warnings: a new connection object is created in `init_server_conn()`.